test/functional/adb-commands-e2e-specs.js

```diff
@@ -236,4 +236,23 @@ describe('adb commands', function () {
       await adb.push(localFile, remoteFile).should.be.rejectedWith(/\/foo\/bar\/remote.txt/);
     });
   });
+
+  describe('bugreport', function () {
+    it('should return the report as raw string by default', async function () {
+      (await adb.bugreport()).should.not.be.empty;
+    });
+
+    it('should return the report as base64-encoded .zip archive if needed', async function () {
+      if (await adb.getApiLevel() < 23) {
```

Comment:
Maybe it's better to move this condition into `bugreport` method, if zipped report is unavailable for API<23?

Reply:
looks like it depends on tools version rather than API version. I'll change the condition

Reply:
I do not think that this is related to tools.

```adb bugreport myreport.zip```

Emu API 23 output:
>Failed to get bugreportz version: 'bugreportz -v' returned '/system/bin/sh: bugreportz: not found' (code 0).
If the device does not run Android 7.0 or above, try 'adb bugreport' instead.

Real Device API 21&22:
>Bugreport is in progress and it could take minutes to complete.
Please be patient and do not cancel or disconnect your device until it completes.
Unexpected string (/system/bin/sh: bugreportz: not found) returned by bugreportz, device probably does not support -z option